### DISCUSSION TITLE
Other Modes of Interacting - vscode in particular

### DISCUSSION THREAD

**Discussion by gordonwatts** [CoffeaTeam/coffea-casa#274] [Q&A]
Does anyone use `vscode` in a meaninful manager with `coffe-casa`?

I'll start by saying I'm a huge fan of `vscode`. I'm not 100% sure how much of the physics community is in my bucket, but it isn't small given the number of twiki help pages for doing ATLAS software development with the tool.

There are a few things that work in `vscode` better than from the command line. As examples:

* Jupyter Notebook interface
* Integrated Python testing (pytest)
* Nicer terminal window
* Python predictive typing based on a type model for both python and notebook
* git GUI interactions (so much nicer than add/commit/push!).
* GUI interaction and local editing often means that latency due to being located a long internet-distance away is hidden.

`vscode` is quite happy to run on remote locations - I regularly use an ssh tunnel and from my house in the south of France run on a largish desktop sitting in our offices at UW. It is also quite happy to build containers and run inside them (it has a whole scheme to build an environment).

**Comment by gordonwatts** [comment]
@oshadura - you mentioned at the meeting someone was already doing this - could you ping them so they answer here?

**Comment by oshadura** [comment]
I used via connecting  directly to a JupyterHub from Visual Studio Code: https://blog.jupyter.org/connect-to-a-jupyterhub-from-visual-studio-code-ed7ed3a31bcb

Is it something you are looking for?

**Comment by gordonwatts** [comment]
That is what I wanted to do - but I didn't know how to get things like the access token without seeing the server log - is that available somewhere?
